README.md

```diff
@@ -1,222 +1,515 @@
-  doGetAndCallback: function(callback){
-    var that = this;
-    $.get('some/ajax/request', function(){
-      callback(that.message);
-    });
-  }
+export default function (foo, bar) {
+  // Do something
 };
 ```
-**Do either of these instead:**
+
+**Do this instead:**
+
 ```javascript
-// Save just the info you need
-var MyModule = {
-  message: 'Hello World!',
-  doGetAndCallback: function(callback){
-    var message = this.message;
-    $.get('some/ajax/request', function(){
-      callback(message);
-    });
-  }
+export const doSomething = function(foo, bar) {
+  // Do something
 };
 
-// Change the scope to make sure it is called in the proper context
-var MyModule = {
-  message: 'Hello World!',
-  doGetAndCallback: function(callback){
-    $.get('some/ajax/request', _.bind(function(){
-      callback(this.message);
-    }, this));
-  }
+export default {
+  doSomething
 };
 ```
-**Or, better yet, return a Deferred/Promise, and let the caller setup the callback on their end:**
+
+#### Importing
+
+Imports should be listed in alphabetical order by their assignment.
+Local imports should be listed separately from external imports.
+
+**Don't do this:**
+
 ```javascript
-var MyModule={
-  message: 'Hello World!',
-  doGet: function(){
-    return $.get('some/ajax/request');
-  }
-};
+import moment from `moment`;
+import foo from './foo';
+import lodash from `lodash`;
+
+
+const foo = 'bar';
 ```
 
-## Curly Braces
+**Do this instead:**
+
+```javascript
+import lodash from `lodash`;
+import moment from `moment`;
+
+import foo from './foo';
 
-Curly braces are required around blocks in loops and conditionals. This is [enforced by jshint](http://www.jshint.com/docs/options/#curly).
+
+const foo = 'bar';
+```
+
+The members of module imports shouldn't be detached from their parent objects. This allows for mocks and stubs to be used during testing, and may allow for easier refactoring.
 
 **Don't do this:**
+
 ```javascript
-if(foo)
-  console.log('bar');
+import {actions, requests} from `service-client`;
+
+
+const {fetchFoo} = requests;
+const {calculateBar} = actions;
+
+let bar = fetchFoo().then(calculateBar);
 ```
+
 **Do this instead:**
+
 ```javascript
-if(foo){
-  console.log('bar');
+import {actions, requests} from `service-client`;
+
+
+let bar = requests.fetchFoo()
+  .then(actions.calculateBar);
+```
+
+### Promises
+
+Asynchronous operations should be written using promises.
+
+* Libraries reliant on promises should adhere to the [Promises/A+ standard]([Promises](http://promises-aplus.github.io/promises-spec/)).
+* [Bluebird](http://bluebirdjs.com) is Yola's promise implementation of choice.
+* ["You're missing the point of promises"](http://domenic.me/2012/10/14/youre-missing-the-point-of-promises/) by Domenic Denicola
+  - A valued article discussing the philosophy behind promises
+
+### JSDoc
+
+JSDoc shouldn't be used. Rules referring to JSDoc usage in the Google JavaScript Style Guide should be ignored.
```

Comment:
This isn't what I communicated here: https://github.com/yola/production/issues/3013#issuecomment-190867562